terraform/modules/kubernetes/monitoring.tf

```diff
@@ -115,3 +115,33 @@ resource "azurerm_monitor_metric_alert" "redis_memory" {
     ]
   }
 }
+
+data "azurerm_monitor_diagnostic_categories" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  resource_id = azurerm_postgresql_flexible_server.postgres-server[0].id
+}
+
+resource "azurerm_log_analytics_workspace" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-logs"
```

Comment:
Can we use suffix "-log" as per https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations